index.bs

```diff
@@ -357,10 +355,46 @@ execute the following steps:
 5.  [$Set the `Sec-CH-UA` header for a request$], given |r|.
 
 
-IDL {#idl} 
-===
+Interface {#interface} 
+=================
+
+<pre class="idl">
+[Exposed=Window]
+interface NavigatorUAData {
+  readonly attribute DOMString brand;
+  readonly attribute DOMString version;
+  readonly attribute DOMString platform;
+  readonly attribute DOMString architecture;
+  readonly attribute DOMString model;
+  readonly attribute boolean mobile;
+};
+interface mixin NavigatorUA {
+  [SecureContext] Promise&lt;NavigatorUAData&gt; getUserAgent();
+};
+Navigator includes NavigatorUA;
+
+</pre>
 
-ISSUE: Yoav should write all this.
+Processing model {#processing}
+--------------
+<dfn method for="NavigatorUA"><code>getUserAgent()</code></dfn> method MUST run these steps:
+
+1. Let |UAData| be a new {{NavigatorUAData}} object.
+2. Set |UAData|'s {{NavigatorUAData/brand}} to the value of the user agent's [=user agent/brand=].
+3. Set |UAData|'s {{NavigatorUAData/platform}} to the value of the user agent's
+    [=user agent/platform brand and version=].
+4. Set |UAData|'s {{NavigatorUAData/architecture}} to the value of the user
+    agent's [=user agent/platform architecture=].
+5. Set |UAData|'s {{NavigatorUAData/model}} to the value of the user agent's [=user agent/model=].
+6. Set |UAData|'s {{NavigatorUAData/mobile}} to the value of the user agent's
+    [=user agent/mobileness=].
+7. Let |version| be the value of the user agent's [=user agent/full version=].
+8. Set |UAData|'s {{NavigatorUAData/version}} to the value of |version|.
+9. Let |promise| be [=a new promise=].
+10. Resolve |promise| with |UAData| [=in parallel=].
+11. Return |promise|.
```

Comment:
I think this could benefit from a little restructuring, like:

```
1.  Let |p| be a [=a new promise=].

2.  Run the following steps [=in parallel=]:

    1.  Let |UAData| be a new {{NavigatorUAData}} object whose values are initialized as follows:

        :   {{NavigatorUAData/brand}}
        ::  The user agent's [=user agent/brand=].
        :   {{NavigatorUAData/platform}}
        ::  The user agent's [=user agent/platform brand and version=].
        ...
        :   {{NavigatorUAData/version}}
        ::  The user agent's [=user agent/full version=].

    2.  [=Resolve=] |promise| with |UAData|.

3.  Return |p|.
```

Reply:
Done!